carma_ros2_utils/README.md

```diff
@@ -41,6 +43,71 @@ public:
 };
 ```
 
+For some nodes you may want to user parameter callbacks to enable dynamic reconfiguring of the node.
+The process for doing this can be a bit tedious so a helper method ```update_params``` with 2 overloads is provided to make it easier.
+The following is an example of how to do this.
+
+```c++
+#include <carma_ros2_utils/carma_lifecycle_node.hpp>
+
+class CarmaLifecycleNodeTest : public carma_ros2_utils::CarmaLifecycleNode
+{
+private:
+  double my_private_param_ = 1.0; // Parameter behind set/get methods to update
+
+public:
+
+  int my_param_ = 0; // A public parameter to update
+
+  // Method takes the new param value and returns the old one
+  double set_my_private_param(double new_val) {
+    double old = my_private_param_;
+    my_private_param_ = new_val;
+    return old;
+  }
+
+  CarmaLifecycleNodeTest(const rclcpp::NodeOptions &options)
+      : CarmaLifecycleNode(options) {}
+
+  ~CarmaLifecycleNodeTest() {};
+
+  // When configuring we want to set up our parameter callback
+  carma_ros2_utils::CallbackReturn handle_on_configure(const rclcpp_lifecycle::State & /*state*/) override
+  {
+
+    // Initial declaration and load of the parameters
+    my_param_ = this->declare_parameter<int>("my_param", my_param_);
+    my_param_ = this->declare_parameter<int>("my_private_param", my_private_param_);
```

Comment:
Since this is working as a template, is there a difference between my_param and my_private_param? The naming makes it seem like one my_param is global

Reply:
Private here refers to class private not ros scope. I'm just trying to show how variables stored in two different contexts could be loaded. It does look like I've assigned this to the wrong one though. It should be  my_private_param_ =